Sources/InhabitableTypeGuards.swift

```diff
@@ -0,0 +1,127 @@
+
+	@discardableResult
+	@available(*, deprecated, message:"Use `promoteValue` instead.")
+	public func flatMap<Inner: SignalProducerConvertible>(_ strategy: FlattenStrategy, _ transform: @escaping (Value) -> Inner) -> Signal<Inner.Value, Error> where Inner.Error == NoError { fatalError() }
+
+}
+
+extension Signal where Value == Never, Error == NoError {
+	@discardableResult
+	@available(*, deprecated, message:"Use `promoteValue` instead.")
+	public func flatMap<Inner: SignalProducerConvertible>(_ strategy: FlattenStrategy, _ transform: @escaping (Value) -> Inner) -> Signal<Inner.Value, Inner.Error> { fatalError() }
+
+	@discardableResult
+	@available(*, deprecated, message:"Use `promoteValue` instead.")
+	public func flatMap<Inner: SignalProducerConvertible>(_ strategy: FlattenStrategy, _ transform: @escaping (Value) -> Inner) -> Signal<Inner.Value, Inner.Error> where Inner.Error == Error { fatalError() }
+}
+
+extension Signal where Error == NoError {
+	@discardableResult
+	@available(*, deprecated, message:"Use `promoteError` instead.")
+	public func flatMapError<NewError>(_ transform: @escaping (Error) -> SignalProducer<Value, NewError>) -> Signal<Value, NewError> { fatalError() }
+}
+
+/*
+func test() {
+	SignalProducer<Never, AnyError>.never.startWithResult { _ in }
+	SignalProducer<Never, NoError>.never.startWithResult { _ in }
+	SignalProducer<Any, NoError>.never.startWithFailed { _ in }
+	SignalProducer<Never, NoError>.never.startWithFailed { _ in }
+	Signal<Never, AnyError>.never.observeResult { _ in }
+	Signal<Never, NoError>.never.observeResult { _ in }
+	Signal<Any, NoError>.never.observeFailed { _ in }
+	Signal<Never, NoError>.never.observeFailed { _ in }
+
+	SignalProducer<Never, AnyError>.never.flatMap(.latest) { _ in SignalProducer<Int, AnyError>.empty }
+	SignalProducer<Never, AnyError>.never.flatMap(.latest) { _ in SignalProducer<Int, NoError>.empty }
+	SignalProducer<Never, NoError>.never.flatMap(.latest) { _ in SignalProducer<Int, AnyError>.empty }
+	SignalProducer<Never, NoError>.never.flatMap(.latest) { _ in SignalProducer<Int, NoError>.empty }
+	SignalProducer<Never, NoError>.never.flatMapError { _ in SignalProducer<Never, AnyError>.empty }
+	SignalProducer<Never, NoError>.never.flatMapError { _ in SignalProducer<Never, NoError>.empty }
+
+	Signal<Never, AnyError>.never.flatMap(.latest) { _ in SignalProducer<Int, AnyError>.empty }
+	Signal<Never, AnyError>.never.flatMap(.latest) { _ in SignalProducer<Int, NoError>.empty }
+	Signal<Never, NoError>.never.flatMap(.latest) { _ in SignalProducer<Int, AnyError>.empty }
+	Signal<Never, NoError>.never.flatMap(.latest) { _ in SignalProducer<Int, NoError>.empty }
+	Signal<Never, NoError>.never.flatMapError { _ in SignalProducer<Never, AnyError>.empty }
+	Signal<Never, NoError>.never.flatMapError { _ in SignalProducer<Never, NoError>.empty }
+}
+*/
```

Comment:
Just a bunch of test cases. Leaving them in the spec would flood the warnings. 🤦‍♂️